_source/_posts/2022-08-12-ci-cd-microservices-circleci-spinnaker.md

```diff
@@ -0,0 +1,949 @@
+```
+
+For both applications, choose the following options:
+- What CI/CD pipeline do you want to generate? **CircleCI**
+- What tasks/integrations do you want to include? (none)
+
+You must tweak the generated configuration at `store/.circleci/config.yml`, as the following changes are required for successful execution:
+- Change the execution environment of the workflow, to a dedicated VM, as that is required by the [TestContainers](https://www.testcontainers.org/supported_docker_environment/continuous_integration/circle_ci) dependency in tests. The chosen environment is [LinuxVM](https://circleci.com/developer/machine/image/ubuntu-2004), a standalone Ubuntu 20.04 virtual machine.
+- As the dedicated VM includes docker, the docker installation step in the configuration must be removed.
+- Add a conditional step for building and pushing the image to DockerHub, only if the branch name is `main`.
+
+**Note**: You can also configure the job to only build specific [branches](https://circleci.com/docs/configuration-reference#branches).
+
+
+The final `config.yml` must look like this:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    environment:
+      DOCKERHUB_USER: your-dockerhub-username
+      IMAGE_NAME: your-dockerhub-username/store
+    machine:
+      image: ubuntu-2004:current
+    resource_class: large
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+            # Perform a Partial Cache Restore (https://circleci.com/docs/2.0/caching/#restoring-cache)
+            - v1-dependencies-
+      - run:
+          name: Print Java Version
+          command: 'java -version'
+      - run:
+          name: Print Node Version
+          command: 'node -v'
+      - run:
+          name: Print NPM Version
+          command: 'npm -v'
+      - run:
+          name: Install Node Modules
+          command: 'npm install'
+      - save_cache:
+          paths:
+            - node
+            - node_modules
+            - ~/.gradle
+          key: {%raw%}v1-dependencies-{{ checksum "build.gradle" }}-{{ checksum "package-lock.json" }}{%endraw%}
+      - run:
+          name: Give Executable Power
+          command: 'chmod +x gradlew'
+      - run:
+          name: Backend tests
+          command: npm run ci:backend:test
+      - when:
+          condition:
+            equal: [ main, << pipeline.git.branch >> ]
+          steps:
+            - run:
+                name: Build and Publish Docker Image
+                command: |
+                  ./gradlew -Pprod bootJar jib \
+                    -Djib.to.image=$IMAGE_NAME \
+                    -Djib.to.auth.username=$DOCKERHUB_USER \
+                    -Djib.to.auth.password=$DOCKERHUB_PASS          
+```
+
+Do the same updates to the `gateway/.circleci/config.yml` file. Additionally, the following change is required:
+
+- Add `~/.cache/Cypress` as a cache path
```

Comment:
@indiepopart Is this something that's still required or something I added. I forget.

Reply:
I think the cache is only required for e2e, which was not added.